tests/phpunit/tests/query/cacheResults.php

```diff
@@ -726,6 +726,110 @@ public function test_query_cache_different_fields() {
 		$this->assertNotEquals( $query1->posts, $query2->posts );
 	}
 
+
+	/**
+	 * @ticket 59188
+	 */
+	public function test_query_cache_unprimed_parents() {
+		$args   = array(
+			'cache_results' => true,
+			'fields'        => 'id=>parent',
+		);
+		$query1 = new WP_Query();
+		$query1->query( $args );
+
+		$post_ids = wp_list_pluck( $query1->posts, 'ID' );
+		wp_cache_delete_multiple( $post_ids, 'post_parent' );
+
+		$queries_before = get_num_queries();
+		$query2         = new WP_Query();
+		$query2->query( $args );
+		$queries_after = get_num_queries();
+
+		$this->assertSame( 1, $queries_after - $queries_before );
+		$this->assertCount( 5, $query1->posts );
+		$this->assertCount( 5, $query2->posts );
+		$this->assertSame( $query1->found_posts, $query2->found_posts );
+	}
+
+	/**
+	 * @ticket 59188
+	 */
+	public function test_query_cache_update_parent() {
+		$page_id = self::factory()->post->create(
+			array(
+				'post_type'   => 'page',
+				'post_parent' => self::$pages[0],
+			)
+		);
+		$args    = array(
+			'cache_results' => true,
+			'post_type'     => 'page',
+			'fields'        => 'id=>parent',
+			'post__in'      => array(
+				$page_id,
+			),
+		);
+		$query1  = new WP_Query();
+		$query1->query( $args );
+
+		wp_update_post(
+			array(
+				'ID'          => $page_id,
+				'post_parent' => self::$pages[1],
+			)
+		);
+
+		$queries_before = get_num_queries();
+		$query2         = new WP_Query();
+		$query2->query( $args );
+		$queries_after = get_num_queries();
+
+		$this->assertSame( self::$pages[0], $query1->posts[0]->post_parent );
+		$this->assertSame( self::$pages[1], $query2->posts[0]->post_parent );
+		$this->assertSame( 2, $queries_after - $queries_before );
+		$this->assertSame( $query1->found_posts, $query2->found_posts );
```

Comment:
Out of curiosity, why is helpful to test that both queries return the same number for posts here and it the previous test?